### ISSUE TITLE
`no-matching-overload` false positive on a valid overload when a sibling overload has inconsistent `@staticmethod` usage

### ISSUE THREAD

**Issue by LoicRiegel** [astral-sh/ty#4388]
## Summary

When a class defines an overloaded method where one overload is decorated with `@staticmethod` and another is a regular instance method (an inconsistent overload set per the typing spec), `ty` reports `invalid-overload` for the inconsistency, as expected. In addition, it reports `no-matching-overload` on a call site that, at least to me, looks like it should match the remaining instance-method overload once `self` is bound.

For comparison, mypy also reports the `@staticmethod` consistency error, but does not error on the call site — it still resolves it against the instance-method overload. I'm not sure whether `ty`'s stricter behavior here is intentional (e.g. deciding not to trust any overload once the set is inconsistent) or a resolution bug, so I'm reporting it in case it's the latter.

## Reproduction

```python
# test.pyi
from typing import overload


class Widget:
    @overload
    @staticmethod
    def exec(actions: list[str], pos: int) -> str: ...
    @overload
    def exec(self, pos: int) -> str: ...


w = Widget()
w.exec(5)
```

`ty check test.pyi`:

```
error[invalid-overload]: Overloaded function `exec` does not use the `@staticmethod` decorator consistently
 --> test.pyi:8:9
  |
7 |     @overload
  |     ---------
8 |     def exec(self, pos: int) -> str: ...
  |         ^^^^
  |         |
  |         Missing here

error[no-matching-overload]: No overload of function `Widget.exec` matches arguments
  --> test.pyi:11:1
   |
11 | w.exec(5)
   | ^^^^^^^^^
info: First overload defined here
 --> test.pyi:4:5
  |
4 | /     @overload
5 | |     @staticmethod
6 | |     def exec(actions: list[str], pos: int) -> str: ...
  | |______________________________________________________^ First overload defined here
info: Possible overloads for function `exec`:
info:   (actions: list[str], pos: int) -> str
info:   (self, pos: int) -> str
```

`mypy` on the same file only reports the consistency error, and does not error on `w.exec(5)`:

```
test.pyi:4: error: Overload does not consistently use the "@staticmethod" decorator on all function signatures.  [misc]
        @overload
         ^
Found 1 error in 1 file (checked 1 source file)
```

## Question / expected behavior

Is the `no-matching-overload` on `w.exec(5)` intentional, given the overload set is already flagged as inconsistent? If not, it would be helpful if `ty` still resolved the call against the `(self, pos: int) -> str` overload (as mypy does) instead of, or in addition to, the `invalid-overload` diagnostic.

### Version

ty 0.0.74 (00199f0aa 2026-08-22)

**Comment by carljm** [comment]
I do think the mypy approach is marginally better here for recovery. I don't think a type checker should be required to consider the one overload a static method and the other one not -- that negates the whole purpose of the rule about placement of `@staticmethod`, which is to prevent that case. But I do think that since the `@staticmethod` decorator is placed in an invalid position, it's more reasonable to discard it entirely and treat both overloads as instance methods, than to treat both overloads as staticmethods like we currently do.
